packages/flutter/lib/src/material/slider.dart

```diff
@@ -684,26 +684,41 @@ class _SliderState extends State<Slider> with TickerProviderStateMixin {
   void _actionHandler(_AdjustSliderIntent intent) {
     final _RenderSlider renderSlider = _renderObjectKey.currentContext!.findRenderObject()! as _RenderSlider;
     final TextDirection textDirection = Directionality.of(_renderObjectKey.currentContext!);
+    bool sliderIncreased = false;
+    widget.onChangeStart?.call(widget.value);
+
     switch (intent.type) {
       case _SliderAdjustmentType.right:
         switch (textDirection) {
           case TextDirection.rtl:
             renderSlider.decreaseAction();
           case TextDirection.ltr:
             renderSlider.increaseAction();
+            sliderIncreased = true;
         }
       case _SliderAdjustmentType.left:
         switch (textDirection) {
           case TextDirection.rtl:
             renderSlider.increaseAction();
+            sliderIncreased = true;
           case TextDirection.ltr:
             renderSlider.decreaseAction();
         }
       case _SliderAdjustmentType.up:
         renderSlider.increaseAction();
+        sliderIncreased = true;
       case _SliderAdjustmentType.down:
         renderSlider.decreaseAction();
     }
+
```

Comment:
This is almost completely a matter of style, but I'd consider playing with this if statement and seeing if you could make it cleaner.  I don't like the fact that line 716 and line 720 are almost completely identical (it's hard to parse out the difference).   Some thought would be to lose the if statement and use a ternary operator to distinguish between the increased/decreased case inline in the parameter to campDouble.  Or to pull out the computation of that value to a separate statement and do something like
```
var unit = renderSlider._semanticActionUnit,0.0,1.0);
var change = renderSlider.value + (sliderIncreased ? : unit : - unit);
widget.onChangeEnd?.call(_lerp(clampDouble(change)));
```

Reply:
Noted, I like that, it is more concise.